compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -869,3 +932,70 @@ pub(crate) fn extract_option_generic_type(typ: Type) -> Type {
 
     generics.pop().expect("Expected Option to have a T generic type")
 }
+
+fn hir_pattern_to_tokens(interner: &NodeInterner, hir_pattern: &HirPattern) -> Vec<Token> {
+    let mut tokens = Vec::new();
+    gather_hir_pattern_tokens(interner, hir_pattern, &mut tokens);
+    tokens
+}
+
+fn gather_hir_pattern_tokens(
+    interner: &NodeInterner,
+    hir_pattern: &HirPattern,
+    tokens: &mut Vec<Token>,
+) {
+    match hir_pattern {
+        HirPattern::Identifier(hir_ident) => {
+            let name = interner.definition_name(hir_ident.id).to_string();
+            tokens.push(Token::Ident(name));
+        }
+        HirPattern::Mutable(pattern, _) => {
+            tokens.push(Token::Keyword(crate::token::Keyword::Mut));
+            gather_hir_pattern_tokens(interner, pattern, tokens);
+        }
+        HirPattern::Tuple(patterns, _) => {
+            tokens.push(Token::LeftParen);
+            for (index, pattern) in patterns.iter().enumerate() {
+                if index != 0 {
+                    tokens.push(Token::Comma);
+                }
+                gather_hir_pattern_tokens(interner, pattern, tokens);
+            }
+            tokens.push(Token::RightParen);
+        }
+        HirPattern::Struct(typ, fields, _) => {
+            let Type::Struct(struct_type, _) = typ else {
```

Comment:
```suggestion
            let Type::Struct(struct_type, _) = typ.follow_bindings() else {
```
Need to call `follow_bindings` whenever we're matching on one variant of a Type